proxy/proxy.ts

```diff
@@ -0,0 +1,126 @@
+// @ts-ignore
+import isMartianIP from 'martian-cidr'
+import type http from 'node:http'
+import { createServer } from 'node:http'
+import { isIP } from 'node:net'
+import { styleText } from 'node:util'
+
+class BadRequestError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'BadRequestError'
+  }
+}
+
+export const createProxyServer = (
+  config: {
+    fetchTimeout?: number
+    hostnameWhitelist?: string[]
+    isProduction?: boolean
+    productionDomainSuffix?: string
+    silentMode?: boolean
+  } = {}
+): http.Server => {
+  // Main toggle for production features
```

Comment:
The arguments are very simple to explain them in comments (in contrast with different hacks)